src/__init__.py

```diff
@@ -102,23 +88,30 @@ def _check_timepoints(t_ampa_ms, t_gaba_ms, tau_ampa, tau_gaba, t_eval_ms, I_amp
         # ampa has to be equal to or smaller than eval
         # for a range of ampa, if eval is within range, use linear/quadratic interpolation
         # if not, raise exception
+        interp_ampa = None
+
         for t in [np.min(t_eval_ms), np.max(t_eval_ms)]:
             if len(t_ampa_ms) == 1:
                 if t > t_ampa_ms[0]:
                     raise Exception("ampa not valid")
-            elif t < np.min(t_ampa_ms) or t > np.max(t_gaba_ms):
+            elif t - tau_ampa < np.min(t_ampa_ms) or t - tau_ampa > np.max(t_ampa_ms):
                 raise Exception("ampa not valid")
             else:
-                interp_ampa = interp1d(t_ampa_ms, I_ampa)
-                I_ampa_eval = interp1d(t_eval_ms - tau_ampa)
+                interp_ampa = interp1d(t_ampa_ms, ampa, kind= 'quadratic')
+        return interp_ampa
 
             #else:
                 #t_ampa_chosen = interp1d(t_ampa_ms, t_eval_ms, kind = int)
                 #if t > t_ampa_chosen:
                     #raise Exception("ampa not valid")
-
+    def _check_gaba_timepoints(self, gaba, t_gaba_ms, t_eval_ms, tau_gaba) :
```

Comment:
why do you have two separate functions (for ampa and gaba) that do pretty much the same thing? better to have one function that interpolates currents and takes in `tau`